src/uu/hashsum/src/hashsum.rs

```diff
@@ -428,8 +392,37 @@ pub fn uu_app_common<'a>() -> Command<'a> {
         )
 }
 
+pub fn uu_app_b3sum<'a>() -> Command<'a> {
+    uu_app_b3sum_opts(uu_app_common())
+}
+
+fn uu_app_b3sum_opts(command: Command) -> Command {
+    command.arg(
+        Arg::new("no-names")
+            .long("no-names")
+            .help("Omits filenames in the output (option not present in GNU/Coreutils)"),
+    )
+}
+
+pub fn uu_app_bits<'a>() -> Command<'a> {
+    uu_app_opt_bits(uu_app_common())
+}
+
+fn uu_app_opt_bits(command: Command) -> Command {
+    // Needed for variable-length output sums (e.g. SHAKE)
+    command.arg(
+        Arg::new("bits")
+            .long("bits")
```

Comment:
The core motivation for this refactor is that GNU Coreutils has `-l/--length` for specifying the output length, which `b2sum` expects, while `--bits` is an invalid command on any GNU Coreutils `*sum` binary. We could simplify this if we removed this flag and instead used `-l/--length` for `sha3sum` and `shake*sum` (if the resolution to https://github.com/uutils/coreutils/issues/2930#issuecomment-1146706740 is to interpret `-l/--length` as bits), but that would be a breaking change for previous users of `coreutils`.